File: apps/api/src/main/java/com/scalebiometrics/api/entity/Tenant.java
package com.scalebiometrics.api.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tenants", schema = "public")
public class Tenant {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "tenant_id", unique = true, nullable = false)
    private String tenantId;

    @Column(nullable = false)
    private String name;

    private String description;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private TenantStatus status;

    @Column(name = "shard_strategy", nullable = false)
    private String shardStrategy;

    @Column(name = "shard_count")
    private Integer shardCount;

    @Column(name = "max_records")
    private Integer maxRecords;

    @Column(name = "callback_url")
    private String callbackUrl;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "updated_by")
    private String updatedBy;

    public enum TenantStatus {
        PENDING,
        ACTIVE,
        SUSPENDED,
        ARCHIVED
    }
}

File: apps/web/types/tenant.ts
export type TenantStatus = 'ACTIVE' | 'SUSPENDED' | 'PENDING' | 'DELETED';

export type TenantShardStrategy = 'NONE' | 'MODULO' | 'RANGE' | 'GEO';

export interface Tenant {
  id: string;
  tenantId: string;
  name: string;
  description?: string;
  status: TenantStatus;
  shardStrategy: TenantShardStrategy;
  shardCount: number;
  maxRecords: number;
  createdAt: string;
  updatedAt: string;
}

export interface TenantConfig {
  key: string;
  value: string;
  category: string;
}

export interface TenantUsage {
  tenantId: string;
  currentRecords: number;
  storageUsedBytes: number;
  requestsThisMonth: number;
  apiCallsRemaining: number;
}

export interface TenantWithUsage extends Tenant {
  usage: TenantUsage;
}
